### PR TITLE
doc: add test coverage and proposal policy

### PR DESCRIPTION
**Pull request by nashif**
Provide some details about submitting proposals and document that new
major features require tests to be added.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -566,6 +566,34 @@ Other Commit Expectations
   proposals, progress snapshots of your work, or to get early feedback on
   features or changes that will affect multiple areas in the code base.
 
+* When major new functionality is added, tests for the new functionality MUST be
+  added to the automated test suite. All new APIs MUST be documented and tested
+  and test MUST cover at least 80% of the added functionality using the code
```
Comment (dbkinder) [inline CONTRIBUTING.rst]: s/test/tests/  or maybe testing

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -566,6 +566,34 @@ Other Commit Expectations
...
+  coverage tool and reporting provided by the project.
+
+Submitting Proposals
+====================
+
+You can request a new feature or submit a proposal by submitting an issue to
+our GitHub Repository.
+If you would like to implement a new feature, please submit an issue with a
+proposal (RFC) for your work first, to be sure that we can use it. Please
```
Comment (dbkinder) [inline CONTRIBUTING.rst]: maybe mention adding ``RFC:`` to the title of the issue?

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.rst (hunk 2 above)
Comment (nashif) [reply CONTRIBUTING.rst]: detailed howto for proposal will follow.

**Comment by nashif** [comment]
@dbkinder  updated

**Review by dbkinder — APPROVED** [review]

**Pull request merged**